Ensure that std::string is present

Compilers occasionally change what headers include which, in this case
GCC 10 got rid of <string> from some include chain.

Author: Lars Viklund <zao@zao.se>
diff -ru binutils-2.32.orig/gold/errors.h binutils-2.32/gold/errors.h
--- binutils-2.32.orig/gold/errors.h	2019-01-19 17:01:33.000000000 +0100
+++ binutils-2.32/gold/errors.h	2020-05-12 23:54:47.670016324 +0200
@@ -24,6 +24,7 @@
 #define GOLD_ERRORS_H
 
 #include <cstdarg>
+#include <string>
 
 #include "gold-threads.h"
 
